tests/deploy/test_scheduler_agent.py

```diff
@@ -210,18 +210,20 @@ async def join(self, thread_pool_finalizer: list[ThreadPoolExecutor], timeout: f
         pass
 
 
-state_translation_table: dict[const.ResourceState, Tuple[state.DeploymentResult, state.BlockedStatus, state.ResourceStatus]] = {
+state_translation_table: dict[
+    const.ResourceState, Tuple[state.DeploymentResult, state.BlockedStatus, state.ComplianceStatus]
+] = {
     # A table to translate the old states into the new states
     # None means don't care, mostly used for values we can't derive from the old state
-    const.ResourceState.unavailable: (None, state.BlockedStatus.NO, state.ResourceStatus.UNDEFINED),
+    const.ResourceState.unavailable: (None, state.BlockedStatus.NO, state.ComplianceStatus.UNDEFINED),
```

Comment:
I think this one too, but less sure.
```suggestion
    const.ResourceState.unavailable: (None, state.BlockedStatus.NO, state.ComplianceStatus.FAILED),
```